datastore/unit_tests/test_query.py

```diff
@@ -76,6 +76,12 @@ def test_ctor_explicit(self):
         self.assertEqual(query.order, ORDER)
         self.assertEqual(query.distinct_on, DISTINCT_ON)
 
+    def test_ctor_w_empty_client(self):
+        _PROJECT = 'OTHER_PROJECT'
+        query = self._makeOne(client=None, project=_PROJECT)
+        self.assertIsNone(query._client)
+        self.assertIsNone(query.namespace)
```

Comment:
Add an assertion that `PROJECT` is correctly assigned in this case.

Also, we will need another testcase which exercises the error path (both `client` and `project` are `None`).